runtime/gc_glue_java/CollectorLanguageInterfaceImpl.cpp

```diff
@@ -700,13 +700,35 @@ MM_CollectorLanguageInterfaceImpl::scavenger_switchConcurrentForThread(MM_Enviro
 			j9tty_printf(PORTLIB, "%p: Nursery [%p,%p] Evacuate [%p,%p] GS [%p,%p] Section size 0x%zx, sections %lu bit offset %lu bit mask 0x%zx\n",
 					vmThread, nurseryBase, nurseryTop, gsBase, gsTop, pageBase, pageTop, _extensions->getConcurrentScavengerPageSectionSize() , sectionCount, startOffsetInBits, bitMask);
 		}
-		vmThread->evacuateBase = gsBase;
-		vmThread->evacuateTop = gsTop;
+		/*
+		 * vmThread->evacuateTop is defined as the last address possible in the evacuate region;
+		 * however, gsTop is the first address above the evacuate region;
+		 * therefore, vmThread->evacuateTop = gsTop - 1.
+		 * In short, the evacuate region is [vmThread->evacuateBase, vmThread->evacuateTop].
+		 */
+		vmThread->evacuateBase = (UDATA)gsBase;
+		vmThread->evacuateTop = (UDATA)gsTop - 1;
```

Comment:
I understand that this line has been modified after I approved the PR. I approve this additional change, too. 

This change is necessary for CR config in case gsTop is at highest possible for CR range, in which case evacuateTopCompressed  would be 0x100000000 after the shift is applied, which would not fit into a 32bit variable/register. Like this highest possible evacuateTopCompressed would be 0xffffffff.